File: kdevelop/languages/ruby/app_templates/kxt/appview.rb

=begin
  This is the main view class for %{APPNAME}.  Most of the non-menu,
  non-toolbar, and non-statusbar (e.g., non frame) GUI code should go
  here.
 
  This %{APPNAMELC} uses an HTML component as an example.
 
  @short Main view
  @author %{AUTHOR} <%{EMAIL}>
  @version %{VERSION}
=end
class %{APPNAMESC}View < %{APPNAMESC}view_base

    #
    # Use this signal to change the content of the statusbar
    #
    signals 'signalChangeStatusbar(const QString&)'

    #
    # Use this signal to change the content of the caption
    #
    signals 'signalChangeCaption(const QString&)'

    slots 'switchColors()',
          'settingsChanged()'
   
    def initialize(parent)
        super(parent)
        
        settingsChanged()
    end
    
    def switchColors()
        color = Settings.instance.col_background
        Settings.instance.setCol_background( Settings.instance.col_foreground )
        Settings.instance.setCol_foreground( color )

        settingsChanged()
    end

    def settingsChanged()
        sillyLabel.setPaletteBackgroundColor( Settings.instance.col_background )
        sillyLabel.setPaletteForegroundColor( Settings.instance.col_foreground )
        # i18n : internationallization
        sillyLabel.setText( i18n("This project is %d days old" % Settings.instance.val_time) )
        
        emit signalChangeStatusbar( i18n("Settings changed") )
    end
        
    def slotSetTitle(title)
        emit signalChangeCaption(title)
    end

end

File: languages/cpp/app_templates/kxt/appview.cpp
%{CPP_TEMPLATE}

#include "%{APPNAMELC}view.h"
#include "settings.h"

#include <klocale.h>
#include <qlabel.h>

%{APPNAMELC}View::%{APPNAMELC}View(QWidget *parent)
    : %{APPNAMELC}view_base(parent)
{
	settingsChanged();
}

%{APPNAMELC}View::~%{APPNAMELC}View()
{

}

void %{APPNAMELC}View::switchColors()
{
	// switch the foreground/background colors of the label
	QColor color = Settings::col_background();
	Settings::setCol_background( Settings::col_foreground() );
	Settings::setCol_foreground( color );

	settingsChanged();
}

void %{APPNAMELC}View::settingsChanged()
{
	sillyLabel->setPaletteBackgroundColor( Settings::col_background() );
	sillyLabel->setPaletteForegroundColor( Settings::col_foreground() );

	// i18n : internationalization
	sillyLabel->setText( i18n("This project is %1 days old", Settings::val_time()) );
	emit signalChangeStatusbar( i18n("Settings changed") );
}

#include "%{APPNAMELC}view.moc"

